damnit/ctxsupport/ctxrunner.py

```diff
@@ -336,10 +258,10 @@ def execute(self, run_data, run_number, proposal, input_vars) -> 'Results':
 
                     # Mymdc fields
                     elif annotation.startswith("mymdc#"):
-                        if mymdc is None:
-                            mymdc = MyMetadataClient(proposal)
-                        metadata = annotation.removeprefix('mymdc#')
-                        kwargs[arg_name] = getattr(mymdc, metadata)(run_number)
+                        if metadata is None:
+                            metadata = extra_proposal.Proposal(proposal)[run_number]
+                        field = annotation.removeprefix('mymdc#')
+                        kwargs[arg_name] = getattr(metadata, field)()
```

Comment:
Not sure how I feel about this in hindsight, it does feel a bit weird to allow calling arbitrary functions. Shall we restrict it to supported fields?

Reply:
It's already enforced by the [check](https://github.com/European-XFEL/DAMNIT/blob/master/damnit/ctxsupport/ctxrunner.py#L184) step when loading the context file.
I can make it explicit here as well if you feel strong about it.